test/cadet/accounts/accounts_test.exs

```diff
@@ -0,0 +1,147 @@
+defmodule Cadet.AccountsTest do
+  @moduledoc """
+  Some tests in this module use pre-recorded HTTP responses saved by ExVCR.
+  this allows testing without the use of actual external IVLE API calls.
+
+  In the case that you need to change the recorded responses, you will need
+  to set the two environment variables IVLE_KEY (used as a module attribute in
+  `Cadet.Accounts.IVLE`) and TOKEN (used here). Don't forget to delete the
+  cassette files, otherwise ExVCR will not override the cassettes.
+  """
+
+  use Cadet.DataCase
+  use ExVCR.Mock, adapter: ExVCR.Adapter.Hackney
+
+  alias Cadet.Accounts
+  alias Cadet.Repo
+  alias Cadet.Accounts.Query
+
+  @token String.replace(inspect(System.get_env("TOKEN")), ~s("), "")
+
+  setup_all do
+    HTTPoison.start()
+  end
+
+  test "create user" do
+    {:ok, user} =
+      Accounts.create_user(%{
+        name: "happy user",
+        role: :student
+      })
+
+    assert user.name == "happy user"
+    assert user.role == :student
+  end
+
+  test "invalid create user" do
+    {:error, changeset} =
+      Accounts.create_user(%{
+        name: "happy user",
+        role: :unknown
+      })
+
+    assert %{role: ["is invalid"]} = errors_on(changeset)
+  end
+
+  test "get existing user" do
+    user = insert(:user, name: "Teddy")
+    result = Accounts.get_user(user.id)
+    assert result.name == "Teddy"
+  end
+
+  test "get unknown user" do
+    refute Accounts.get_user(10000)
+  end
+
+  test "associate nusnet_id to user" do
+    user = insert(:user)
+    {:ok, auth} = Accounts.add_nusnet_id(user, "teddy@happy.mail")
+    assert auth.provider == :nusnet_id
+    assert auth.uid == "teddy@happy.mail"
+    assert auth.user_id == user.id
+  end
+
+  test "duplicate nusnet_id one user" do
+    user = insert(:user)
+    {:ok, _} = Accounts.add_nusnet_id(user, "teddy@happy.mail")
+    {:error, changeset} = Accounts.add_nusnet_id(user, "teddy@happy.mail")
+    assert %{uid: ["has already been taken"]} = errors_on(changeset)
+  end
+
+  test "duplicate nusnet_id different user" do
+    user = insert(:user)
+    {:ok, _} = Accounts.add_nusnet_id(user, "teddy@happy.mail")
+    user2 = insert(:user)
+    {:error, changeset} = Accounts.add_nusnet_id(user2, "teddy@happy.mail")
+    assert %{uid: ["has already been taken"]} = errors_on(changeset)
+  end
+
+  test "setting user nusnet_id without e-mail" do
+    user = insert(:user)
+    assert {:ok, []} = Accounts.set_nusnet_id(user, "E012345")
+  end
+
+  # TODO: A user may not have multiple NUSNET_IDs?
```

Comment:
I think it's a good idea to validate against this, but probably can be done in a separate PR. This PR is very big already haha

Reply:
Yep. Didn't know that this TODO would get pass credo though. Guess it ignores the test files.